pints/tests/test_dual_averaging.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+#
+# Tests the basic methods of the Hamiltonian MCMC routine.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import numpy as np
+import unittest
+
+import pints
+
+
+class TestDualAveraging(unittest.TestCase):
+    """
+    Tests the dual averaging adaption used in the NUTS sampler.
+    """
+
+    def test_dual_averaging(self):
```

Comment:
Can this be split up into small chunks?
If not, please add some comments at least, showing what is being tested